pkg/pool-weighted/contracts/smart/LiquidityBootstrappingPool.sol

```diff
@@ -59,73 +85,345 @@ contract LiquidityBootstrappingPool is BaseWeightedPool, ReentrancyGuard {
             owner
         )
     {
-        uint256 numTokens = tokens.length;
-        _require(numTokens <= _MAX_LBP_TOKENS, Errors.MAX_TOKENS);
+        uint256 totalTokens = tokens.length;
+        _require(totalTokens <= _MAX_LBP_TOKENS, Errors.MAX_TOKENS);
+
+        InputHelpers.ensureInputLengthMatch(totalTokens, normalizedWeights.length);
 
-        InputHelpers.ensureInputLengthMatch(numTokens, normalizedWeights.length);
+        bytes32 poolState;
 
-        // Ensure each normalized weight is above the minimum and find the token index of the maximum weight
+        // Ensure valid weights
         uint256 normalizedSum = 0;
-        uint256 maxWeightTokenIndex = 0;
-        uint256 maxNormalizedWeight = 0;
-        for (uint8 i = 0; i < numTokens; i++) {
+        for (uint8 i = 0; i < totalTokens; i++) {
             uint256 normalizedWeight = normalizedWeights[i];
             _require(normalizedWeight >= _MIN_WEIGHT, Errors.MIN_WEIGHT);
 
-            _setNormalizedWeight(normalizedWeight, i);
+            // Insert "start weights" and "end weights" into poolState
+            // Before the first update, start=end, which avoids initialization edge cases
+            poolState = poolState.insertUint32(normalizedWeight.compress32(), i * 32);
+            poolState = poolState.insertUint16(normalizedWeight.compress16(), _END_WEIGHT_OFFSET + i * 16);
 
             normalizedSum = normalizedSum.add(normalizedWeight);
-            if (normalizedWeight > maxNormalizedWeight) {
-                maxWeightTokenIndex = i;
-                maxNormalizedWeight = normalizedWeight;
-            }
         }
         // Ensure that the normalized weights sum to ONE
         _require(normalizedSum == FixedPoint.ONE, Errors.NORMALIZED_WEIGHT_INVARIANT);
 
-        // Initial value; can change later if weights change
-        _maxWeightTokenIndex = maxWeightTokenIndex;
+        // Write initial pool state - all zeros except for the start weights
```

Comment:
```suggestion
        // Write initial pool state
```

We now fill in both start and end weights so this is stale. I don't think we need to describe what the state looks like as it isn't an atypical state.

Reply:
Good catch!